src/main/java/org/folio/edge/sip2/repositories/CirculationRepository.java

```diff
@@ -198,12 +312,97 @@ public String getPath() {
   private class CheckoutRequestData extends CirculationRequestData {
     private CheckoutRequestData(JsonObject body, Map<String, String> headers,
         SessionData sessionData) {
-      super(body, headers, sessionData);
+      super(body, null, null, headers, sessionData);
     }
 
     @Override
     public String getPath() {
       return "/circulation/check-out-by-barcode";
     }
   }
+
+  private class RequestsRequestData extends CirculationRequestData {
+    private final String idField;
+    private final String idValue;
+    private final String requestType;
+
+    private RequestsRequestData(
+        String idField,
+        String idValue,
+        String requestType,
+        Integer startItem,
+        Integer endItem,
+        Map<String, String> headers,
+        SessionData sessionData) {
+      super(null, startItem, endItem, headers, sessionData);
+      this.idField = idField;
+      this.idValue = idValue;
+      this.requestType = requestType;
+    }
+
+    @Override
+    public String getPath() {
+      final StringBuilder sb = new StringBuilder()
+          .append("/circulation/requests?query=%28")
+          .append(idField)
+          .append("%3D%3D")
+          .append(idValue)
+          .append("%20and%20status%3DOpen");
+      if (requestType != null) {
+        sb.append("%20and%20requestType%3D%3D")
+          .append(requestType);
+      }
+      sb.append("%29");
```

Comment:
Probably. The vertx web client may do it for us. I just grabbed the URLs from mod-patron and they were hand encoded, for whatever reason. I'll add it to the TODO list. 😄

Reply:
so if the web client may do it for us, then it would double encode the query string, no?  Have you tested end-to-end to see if it would work? 

If there's only one PR comment that you could address due to time constraint, I'd love to see this one being addressed because it's a sore sight to see hard-coded URL encoded strings. Not only that, it would be difficult to maintain/debug. But if there is bigger fish to fry at this moment, this can wait. My two cents  :)

Reply:
It may be a flag or it might autodetect non-URL chars an encode them, leaving the existing encoding in place or it might do none of these. I have tested end to end and that works. I'll take a look into this next.

Reply:
It turns out vertx uses netty to manage the url encoding of parameters, so we don't need to pre-encode, like the code mentioned here. The netty encoder takes care of the '+' encoding for spaces that the Java url encoder uses by replacing them with %20. I've removed the encoding in the strings and things are working as expected.